src/custom/utils/operator/index.ts

```diff
@@ -154,68 +159,80 @@ function toApiAddress(address: string, chainId: ChainId): string {
-  let json
-  try {
-    response = await _fetchGet(chainId, url)
-    json = await response.json()
-  } finally {
-    const genericError = new Error(`Error getting query of type ${type} @ ${url}`)
-    if (!response || !json) {
-      throw genericError
-    } else if (!response.ok) {
-      // is backend error handled at this point
-      const errorResponse: ApiErrorObject = json
-      let error: QuoteError | OperatorError | Error
-      if (type === 'Order') {
-        error = new OperatorError(errorResponse)
-      } else if (type === 'Quote') {
-        // we need to map the backend error codes to match our own for quotes
-        const mappedError = mapOperatorErrorToQuoteError(errorResponse.errorType)
-        error = new QuoteError(mappedError)
-      } else {
-        // shouldn't get here but best to handle
-        error = genericError
-      }
-      throw error
-    } else {
-      return json
-    }
+
+const UNHANDLED_QUOTE_ERROR: QuoteErrorObject = {
+  errorType: QuoteErrorCodes.UNHANDLED_ERROR,
+  description: QuoteErrorDetails.UNHANDLED_ERROR
+}
+
+const UNHANDLED_ORDER_ERROR: ApiErrorObject = {
+  errorType: ApiErrorCodes.UNHANDLED_ERROR,
+  description: ApiErrorCodeDetails.UNHANDLED_ERROR
+}
+
+async function _handleQuoteResponse(response: Response) {
+  if (!response.ok) {
+    const responseNotOkJson: ApiErrorObject = await response.json()
+    const errorType = responseNotOkJson.errorType
+
+    // we need to map the backend error codes to match our own for quotes
+    const mappedError = mapOperatorErrorToQuoteError(errorType)
+    throw new QuoteError(mappedError)
+  } else {
+    return response.json()
   }
 }
 
 export async function getPriceQuote(params: PriceQuoteParams): Promise<PriceInformation> {
   const { baseToken, quoteToken, amount, kind, chainId } = params
   const [checkedBaseToken, checkedQuoteToken] = [checkIfEther(baseToken, chainId), checkIfEther(quoteToken, chainId)]
-  console.log('[util:operator] Get Price from API', params)
+  console.log('[util:operator] Get price from API', params)
 
-  return _getJson(
+  const response = await _fetchGet(
     chainId,
-    `/markets/${toApiAddress(checkedBaseToken, chainId)}-${toApiAddress(checkedQuoteToken, chainId)}/${kind}/${amount}`,
-    'Quote'
-  )
+    `/markets/${toApiAddress(checkedBaseToken, chainId)}-${toApiAddress(checkedQuoteToken, chainId)}/${kind}/${amount}`
+  ).catch(error => {
+    console.error('Error getting price quote:', error)
+    throw new QuoteError(UNHANDLED_QUOTE_ERROR)
+  })
+
+  return _handleQuoteResponse(response)
 }
 
 export async function getFeeQuote(params: FeeQuoteParams): Promise<FeeInformation> {
   const { sellToken, buyToken, amount, kind, chainId } = params
   const [checkedSellAddress, checkedBuyAddress] = [checkIfEther(sellToken, chainId), checkIfEther(buyToken, chainId)]
   console.log('[util:operator] Get fee from API', params)
 
-  return _getJson(
+  const response = await _fetchGet(
```

Comment:
nit, this await / catch / throw, is just a more obscure way of try / catch / throw, isn't it? 

Why not using the latter? 


```ts
try {
    const response = await _fetchGet(
      chainId,
      `/markets/${toApiAddress(checkedBaseToken, chainId)}-${toApiAddress(
        checkedQuoteToken,
        chainId
      )}/${kind}/${amount}`
    )
    return _handleQuoteResponse(response)
  } catch (error) {
    console.error('Error getting price quote:', error)
    throw new QuoteError(UNHANDLED_QUOTE_ERROR)
  }
```

Reply:
what do you mean? that's what it was before where `_handleQuoteResponse` also throws and all errors were being turned into unhandled quote errors. 

the inline way is to catch both separately

Reply:
https://github.com/gnosis/cowswap/pull/770#discussion_r652686520

Reply:
going to merge as this is resolved from your previous comment linked above (i think) and there are 3 approvals

Reply:
It's fine. I think that was working cause u don't do `return await _handleQuoteResponse(response)`  u do `return _handleQuoteResponse(response)` 

And since `_handleQuoteResponse` is async, it will return a promise that is rejected (when it throws internally). So ti won't be catched by the catch (as espected).

The only will capture an error in `await _fetchGet`

Reply:
Good eitherway